x-pack/plugins/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -51,6 +64,23 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
     this.provider = this.config.apiProvider;
     this.key = this.secrets.apiKey;
 
+    this.openAI =
+      this.config.apiProvider === OpenAiProviderType.AzureAi
+        ? new OpenAI({
+            apiKey: this.secrets.apiKey,
+            baseURL: this.config.apiUrl,
+            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
+            defaultHeaders: { 'api-key': this.secrets.apiKey },
+          })
+        : new OpenAI({
```

Comment:
Are we no longer using the URL passed to the OpenAI connector?

Reply:
not when the `invokeAsyncIterator` subaction is used, the rest of the subactions still use the API urls. However, I do want to go further and abandon the API methods and instead use the OpenAI sdk since it handles and standardizes the response. I think it would reduce risk

Reply:
Good catch Dario! The baseURL was defined for azure but not OpenAI. I added it in this commit https://github.com/elastic/kibana/pull/174126/commits/6f3eebfc57f7000b0e76cdf25644b2c60e5471fa#diff-ec8aad2275f18b983a14eecb25411a6b024a3aec3fba3ef965f9ed81802afb68R76